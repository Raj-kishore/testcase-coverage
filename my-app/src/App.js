import logo from './logo.svg';
import './App.css';
import { useEffect, useState } from 'react';

function App() {

  const [result, setResult] = useState(0);
  const [count, setCount] = useState(0);

  function dummyCalculation(a, b) {
    if (a > b) {
      setResult((a + b) * 2);
    }
    setResult(a + b);
  }

  useEffect(() => {
    if (result !== 0) {
      setCount(count + 1);
    }
  }, [result]);

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <button onClick={() => dummyCalculation(14, 6)}>Test1</button>
        <button onClick={() => dummyCalculation(4, 6)}>Test2</button>
        <div id="trackResult">{result}</div>

        {(count > 5) ?
          <div id="stopit">Stop it...</div> :
          <div id="goahead">Go ahead</div>
        }

      </header>
    </div>
  );
}

export default App;
