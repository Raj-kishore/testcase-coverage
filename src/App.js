import React, { Component } from 'react';
import PropTypes from 'prop-types';


import Highcharts from 'highcharts'
import HighchartsReact from 'highcharts-react-official'



class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      hcOptions: {}
    }

  }

  componentDidMount() {
    this.updateState();
  }

  updateState() {
    this.setState({
      hcOptions: {
        chart: {
          zoomType: 'xy'
        },
        title: {
          text: 'Average Monthly Precipitation and Temperature in Karasjok, 2021',
          align: 'left'
        },
        subtitle: {
          text: 'Source: ' +
            '<a href="https://www.yr.no/nb/historikk/graf/5-97251/Norge/Troms%20og%20Finnmark/Karasjok/Karasjok?q=2021"' +
            'target="_blank">YR</a>'
        },
        xAxis: [{
          categories: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
            'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
          crosshair: true
        }],
        yAxis: [{ // Primary yAxis
          labels: {
            format: '{value}Â°C',
            style: {
              color: "rgba(0,0,0,0)"
            }
          },
          title: {
            text: 'Temperature',
            style: {
              color: "rgba(0,0,0,0)"
            }
          }
        }, { // Secondary yAxis
          title: {
            text: 'Precipitation',
            style: {
              color: "rgba(0,0,0,0)"
            }
          },
          labels: {
            format: '{value} mm',
            style: {
              color: "rgba(0,0,0,0)"
            }
          },
          opposite: true
        }],
        tooltip: {
          shared: true,
          formatter () { 
          
            if(this.y < 50){
              return  ` Data not available `;
            }else{
              return  `<b>${this.x}</b><br/> ${this.y}`;
            }
          }
          
        },
        legend: {
          align: 'left',
          x: 80,
          verticalAlign: 'top',
          y: 80,
          floating: true,
          backgroundColor:
            Highcharts.defaultOptions.legend.backgroundColor || // theme
            'rgba(255,255,255,0.25)'
        },
        series: {
          name: 'Precipitation',
          type: 'column',
          yAxis: 1,
          data: [10,20,30,40,50,60,70,80,90,100],
          tooltip: {
            valueSuffix: ' mm'
          }

        }
      }
    })
  }

  render() {
    return (
      <div>
        <HighchartsReact
          highcharts={Highcharts}
          options={this.state.hcOptions}
        />
      </div>
    );
  }
}

App.propTypes = {

};

export default App;