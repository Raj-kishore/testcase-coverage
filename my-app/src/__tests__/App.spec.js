import { render, screen, fireEvent } from '@testing-library/react';
import App from '../App';
import renderer from 'react-test-renderer';
import "@testing-library/jest-dom";




it("check Test functions are triggered", () => {
  const {  getByText } = render(<App />)
  fireEvent.click(getByText(/Test1/));
  fireEvent.click(getByText(/Test2/));
  expect(getByText('Go ahead')).toBeTruthy();
  for(var i=0; i < 2; i++){
    fireEvent.click(getByText(/Test1/));
    fireEvent.click(getByText(/Test2/));
  }
  expect(getByText('Stop it...')).toBeTruthy();
});


test('Snapshot testing', () => {
  const tree = renderer
    .create(<App/>)
    .toJSON();
  expect(tree).toMatchSnapshot(); 
});

