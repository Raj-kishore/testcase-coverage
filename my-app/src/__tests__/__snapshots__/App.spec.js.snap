// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <button
      onClick={[Function]}
    >
      Test1
    </button>
    <button
      onClick={[Function]}
    >
      Test2
    </button>
    <div
      id="trackResult"
    >
      0
    </div>
    <div
      id="goahead"
    >
      Go ahead
    </div>
  </header>
</div>
`;
